src/devices/Arduino/samples/ApiChecker/TestCases.cs

```diff
@@ -57,48 +59,20 @@ public static void Run(ArduinoBoard board)
             while (loop);
         }
 
-        private static void TestAnalogCallback(ArduinoBoard board)
+        private static int GetAnalogPin(ArduinoBoard board, int analogChannel)
         {
-            int analogPin = GetAnalogPin1(board);
-            var analogController = board.CreateAnalogController(0);
-            board.SetAnalogPinSamplingInterval(TimeSpan.FromMilliseconds(10));
-            var pin = analogController.OpenPin(analogPin);
-            pin.EnableAnalogValueChangedEvent(null, 0);
-
-            pin.ValueChanged += (sender, args) =>
-            {
-                if (args.PinNumber == analogPin)
-                {
-                    Console.WriteLine($"New voltage: {args.Value}.");
-                }
-            };
-
-            Console.WriteLine("Waiting for changes on the analog input");
-            while (!Console.KeyAvailable)
-            {
-                // Nothing to do
-                Thread.Sleep(100);
-            }
-
-            Console.ReadKey();
-            pin.DisableAnalogValueChangedEvent();
-            pin.Dispose();
-            analogController.Dispose();
-        }
-
-        private static int GetAnalogPin1(ArduinoBoard board)
-        {
-            int analogPin = 15;
+            int analogPin;
```

Comment:
Seems like the PR still has non-ImageSharp-related changes. I understand why RgbLedTest is changing in this project, but might be easier to review if these other things are moved to a separate PR.

Reply:
It's hard to fully separate the stuff, because the changes depend on each other. Neither of the changes will build without the other. But I separated it in groups if you look at the individual commits.